Add support for replacement variables in the Snippet Window.

herregroen opened the pull request:
Allows usage of fields added by ACF in the Snippet Window.

The fields can be used with %%cf_{name-of-custom-field}%%

This fixes Yoast/wordpress-seo#5632.

This depends on Yoast/wordpress-seo#7607 as it uses the exposed `ReplaceVar` class but will fail gracefully ( not do anything ) if it's not present.

moorscode commented on js/src/replacevars.js:
```diff
@@ -0,0 +1,51 @@
+
+        replaceVars[field.name] = new ReplaceVar( '%%cf_'+field.name+'%%', content, { source: 'direct' } );
+        YoastSEO.wp.replaceVarsPlugin.addReplacement( replaceVars[field.name] );
+        console.log("Created ReplaceVar for: ", field.name, " with: ", content, replaceVars[field.name]);
```
This should be wrapped with a `config.debug`

moorscode commented on js/src/replacevars.js:
```diff
@@ -0,0 +1,51 @@
+        var content = (field.type === 'wysiwyg') ? jQuery(jQuery.parseHTML(field.content)).text() : field.content;
+
+        replaceVars[field.name].replacement = content;
+        console.log("Updated ReplaceVar for: ", field.name, " with: ", content, replaceVars[field.name]);
```
This should be wrapped with a `config.debug`

moorscode commented on js/src/replacevars.js:
```diff
@@ -0,0 +1,51 @@
+module.exports = {
+    createReplaceVars: createReplaceVars,
+    updateReplaceVars: updateReplaceVars
+};
```
Missing newline

herregroen replied:
Has been fixed.

moorscode commented on js/src/collect/collect.js:
```diff
@@ -22,6 +17,17 @@ Collect.prototype.append = function(data){
         field_data = scraper_store.getScraper(type).scrape(field_data);
     });
 
+    return field_data
```
Missing `;`

kraftner commented:
Having a test for this would be nice.

herregroen commented:
Tests have been written. Bugs discovered in the test have been fixed. Also changed two existing tests that were failing.

* The current class is not on the `#menu-dashboard` but a child.
* The image modal opens on the upload tab. Added code to switch to the Media Library.

moorscode commented:
Acceptance done!
